src/main/kotlin/ecommerce/service/impl/ProductServiceImpl.kt

```diff
@@ -0,0 +1,52 @@
+package ecommerce.service.impl
+
+import ecommerce.dto.ProductRequest
+import ecommerce.dto.ProductResponse
+import ecommerce.exception.ProductCreationException
+import ecommerce.exception.ProductNotFoundException
+import ecommerce.exception.ProductUpdateException
+import ecommerce.model.toDto
+import ecommerce.repository.ProductRepository
+import ecommerce.service.ProductService
+import org.springframework.stereotype.Service
+
+@Service
+class ProductServiceImpl(private val productRepository: ProductRepository) : ProductService {
```

Comment:
Nice work 👍
But I think the interface might be a bit redundant for now, since it currently has only one implementation 👀